standard/expressions.md

```diff
@@ -1000,10 +1000,11 @@ In case the parameter type sequences `{P₁, P₂, ..., Pᵥ}` and `{Q₁, Q₂
 
 #### 11.6.4.4 Better conversion from expression
 
-Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a better conversion than `C₂` if one of the following holds:
+Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a ***better conversion*** than `C₂` if one of the following holds:
 
 - `E` exactly matches `T₁` and `E` does not exactly match `T₂` ([§11.6.4.5](expressions.md#11645-exactly-matching-expression))
 - `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a better conversion target than `T₂` ([§11.6.4.6](expressions.md#11646-better-conversion-target))
+- `E` is a method group ([§11.2](expressions.md#112-expression-classifications)), and `T₁` is compatible ([§19.4](delegates.md#194-delegate-compatibility)) with the single best method from the method group for conversion `C₁`
```

Comment:
We probably need to say that T2 is _not_ compatible

Reply:
I don't think we actually do.

As per the original PR comment:

> The rule doesn't refer to T2/C2, but I don't think it has to...
> because it's valid for both conversions to be better than the other.
> That's already the case in better conversion target, as types T1 and
> T2 could both have implicit conversions to each other.

So basically if E is a method group and both T1 and T2 are compatible with the best single method from the method group conversion, then each is better than the other and it's ambiguous.

Now I'd agree that it's a bit weird for both conversions to be better than each other, but given that that's *already* the case elsewhere, I figured I might as well go with the flow.

But I'm happy to be corrected if I'm missing something :)

Reply:
It looks like you're right, that better conversion target _also_ currently allows two types to both be better than one another. However, I think we should fix that too. I seem to remember having done that before, at some point, but who knows which version of spec out there that was in!

Here's why I think we should not allow two conversions to both be better than the other. In 11.6.4.3 Better Function Member we say:

> Given an argument list `A` with a set of argument expressions `{E₁, E₂, ..., Eᵥ}` and two applicable function members `Mᵥ` and `Mₓ` with parameter types `{P₁, P₂, ..., Pᵥ}` and `{Q₁, Q₂, ..., Qᵥ}`, `Mᵥ` is defined to be a ***better function member*** than `Mₓ` if
>
> - for each argument, the implicit conversion from `Eᵥ` to `Qᵥ` is not better than the implicit conversion from `Eᵥ` to `Pᵥ`, and
> - for at least one argument, the conversion from `Eᵥ` to `Pᵥ` is better than the conversion from `Eᵥ` to `Qᵥ`.

The first bullet will fail incredibly often if a type is better than itself, for instance!

So I think we should fix both this method group case, as I proposed, and also the implicit conversion case of better conversion target.

Reply:
Works for me. I'll accept your suggestion (tomorrow) and try to fix the implicit conversion case at the same time.